src/services/scenes.ts

```diff
@@ -36,29 +42,43 @@ export function getColorByIndex(index: number) {
   return visTheme.getColorByName(visTheme.palette[index % 8]);
 }
 
-export function getLabelOptions(scenObject: SceneObject, allOptions: string[]) {
-  const labelFilters = sceneGraph.lookupVariable(VAR_FILTERS, scenObject);
-  const labelOptions: Array<SelectableValue<string>> = [];
+export function getLabelOptions(sceneObject: SceneObject, allOptions: string[]) {
+  const filteredOptions = filterUsedLabelNames(sceneObject, allOptions);
+  const labelOptions: Array<SelectableValue<string>> = filteredOptions.map((label) => ({
+    label,
+    value: String(label),
+  }));
+
+  const levelOption = [];
+  if (!allOptions.includes('level')) {
+    levelOption.push({ label: 'level', value: 'level' });
+  }
+
+  return [{ label: 'All', value: ALL_VARIABLE_VALUE }, ...levelOption, ...labelOptions];
+}
+
+/**
+ * Given an array of label names, return those that are not already present in the filters.
```

Comment:
```suggestion
 * Given an array of label, or fields names, return those that are not already present in the filters.
```

Reply:
I was also thinking about how to make the function name more inclusive for labels and fields. The only thing I came up with is `filterUsedFilters`, but I don't think that it is more understandable. :(

Reply:
Yeah, it's challenging. Maybe `getUniqueFilters`